listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -46,6 +46,42 @@ export default class ListenCard extends React.Component<
   ListenCardProps,
   ListenCardState
 > {
+  static preciseTimestamp = (listen: Listen): string => {
+    const listened_at = listen.listened_at_iso || listen.listened_at * 1000;
+    const listenDate: Date = new Date(listened_at);
+    const msDifference = new Date().getTime() - listenDate.getTime();
+    if (
+      // over one year old : show with year
+      msDifference / (1000 * 3600 * 24 * 365) >
+      1
+    ) {
+      return `${listenDate.toLocaleString("en-US", {
```

Comment:
Just asking a question here, should we hard-code this locale, or is there a way of using the browser's default locale?

Reply:
Nice catch, I missed that! 

We should use `.toLocaleString(undefined, {…options})` so that the user's browser's locale is automatically used (while still passing options as a second argument).

Reply:
Thank you for the review! :)

> This is a topic that everyone has an idea about, and there might be some other opinions here, but I think that we should only show "x minutes ago" for listens less than an hour old. Anything else should have a timestamp.

@alastair I considered this too. Personally since I listen and submit less frequently, they are older and so no "x minutes ago" formatted timestamps would appear on my my recent listens page most of the time. 

![image](https://user-images.githubusercontent.com/57575778/110078839-2ad4d380-7d3d-11eb-9264-180334ae79b9.png)

^ Users are still always able to view the exact timestamp for any listen by hovering. (I could also update this to be more similar to the >1 year format for more readability).

I think allowing listens up to 24 hours old (instead of 1 hour old) to be formatted this way makes the page look less formal for users who listen and submit less frequently, but I'd like to hear more opinions as well.